### PR TITLE
Use browser fetch on renderer

### PR DESCRIPTION
**Pull request by jakolehm**
For performance reasons.

### CODE REVIEW

File: src/common/k8s-api/json-api.ts [hunk 1]
```diff
@@ -65,22 +69,33 @@ export class JsonApi<D = JsonApiData, P extends JsonApiParams = JsonApiParams> {
...
-  get<T = D>(path: string, params?: P, reqInit: RequestInit = {}) {
+  get<T = D>(path: string, params?: P, reqInit: RequestInit | NodeRequestInit = {}) {
     return this.request<T>(path, params, { ...reqInit, method: "get" });
   }
 
-  getResponse(path: string, params?: P, init: RequestInit = {}): Promise<Response> {
-    let reqUrl = `${this.config.serverAddress}${this.config.apiBase}${path}`;
-    const reqInit: RequestInit = merge({}, this.reqInit, init);
+  protected getRequestUrl(path: string, useSubdomain = false) {
+    if (window) {
```
Comment (ixrock) [inline src/common/k8s-api/json-api.ts:86]: I guess it's not gonna work in node env. 
You cannot check global variable existence this way, use `typeof` at least:

```
% node -i
Welcome to Node.js v14.17.6.
Type ".help" for more information.
> window
Uncaught ReferenceError: window is not defined
> typeof window
'undefined'
```

File: src/common/k8s-api/kube-object.store.ts [hunk 2]
```diff
@@ -348,7 +348,7 @@ export abstract class KubeObjectStore<T extends KubeObject> extends ItemStore<T>
     const { signal } = abortController;
 
     const callback = (data: IKubeWatchEvent<T>, error: any) => {
-      if (!this.isLoaded || error?.type === "aborted") return;
+      if (!this.isLoaded || error instanceof DOMException || error?.type === "aborted") return;
```
Comment (ixrock) [inline src/common/k8s-api/kube-object.store.ts:351]: Probably just having `error instanceof DOMException` should be enough for aborted requests.
Or `error?.type === "aborted"` could come not only from DOM's `AbortController`?

File: src/main/lens-proxy.ts [hunk 3]
```diff
@@ -185,15 +185,25 @@ export class LensProxy extends Singleton {
...
   protected async handleRequest(req: http.IncomingMessage, res: http.ServerResponse) {
+    // allow to fetch apis in "clusterId.localhost:port" from "localhost:port"
+    // this should be safe because we have already validated cluster uuid
+        
+    res.setHeader("Access-Control-Allow-Origin", "*");
+    res.setHeader("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, PATCH, OPTIONS");
+    res.setHeader("Access-Control-Allow-Headers", "Content-Type, X-Cluster-Id");
+
+    if (req.method === "OPTIONS") {
+      res.end();
```
Comment (ixrock) [inline src/main/lens-proxy.ts:196]: What the point to allow `OPTIONS` from `Access-Control-Allow-Methods` to end it right after?

Comment (jakolehm) [reply src/main/lens-proxy.ts:196]: Renderer (browser) asks CORS via `OPTIONS`, we just need to return headers and nothing else.

**Comment by jakolehm** [comment]
Closing this because it seems that node-fetch seems to be ok (it was logger causing performance issues).

**Pull request closed without merging**